### DISCUSSION TITLE
Can the DBA Dash repository be shared across multiple installations of DBA Dash on different devices?

### DISCUSSION THREAD

**Discussion by drupalgrupal** [trimble-oss/dba-dash#854] [General]
Can the DBA Dash repository be shared across multiple installations of DBA Dash on different devices?

**Comment by DavidWiseman** [comment]
Yes. You can have multiple services pointing to the same repository database. Each service can collect data from one or more monitored instances. 

It's even possible to have a service write to a s3 bucket rather than directly to the repository database. The S3 bucket is then a source connection for the service that can write to the repository database. This is useful in situations where there isn't direct connectivity between the monitored instance and the repository DB.

Hope this helps

**Reply by drupalgrupal** [reply]
That is excellent and thank you for the prompt reply!

Just one follow up question: Would everyone/devices connecting to that repository will be able to see all the instances, regardless of which service is collecting the data from the monitored instances?

**Reply by DavidWiseman** [reply]
Yes, users will be able to see all instances.  There is an App role in the repository database that you can add users to which allows them to use the GUI.  The GUI will display all instances.  

There is a separate GUI install that can be deployed to end users.  Users will be prompted to upgrade the GUI when you upgrade the repository.

You can use [tagging](https://dbadash.com/docs/help/tagging/) as a way to group & filter your instances.  This can help teams to focus on the instances they are responsible for or interested in while keeping everything centralized.   

If you wanted to prevent users from seeing certain groups of instances, you would need to use separate repository databases.  It is possible to write to [multiple repository databases](https://dbadash.com/docs/setup/options/#secondary-destinations).  This allows you to have a central repository database and team-specific repositories. Reading once from the monitored instances and writing to multiple repositories.  e.g. Service A writes to repository A & repository C.  Service B writes to repository B & repository C.    Team A connects to repository A and Team B connects to repository B and can only see instances collected by the associated service.  Repository C provides a central view of all the instances.
There are extra storage costs and extra processing costs associated with this approach.  Another option is just to have the Team A and Team B repositories.  The GUI can switch between them with the **Connect** button.

Hope this helps.

**Reply by drupalgrupal** [reply]
Thank you David!

I really appreciate your prompt and detailed replies.
